app/components/SocketProvider.js

```diff
@@ -174,14 +174,19 @@ class SocketProvider extends React.Component<Props> {
           const collection = collections.get(collectionId) || {};
 
           if (event.event === "collections.delete") {
-            documents.removeCollectionDocuments(collectionId);
-            memberships.removeCollectionMemberships(collectionId);
-
             const collection = collections.get(collectionId);
             if (collection) {
               collection.deletedAt = collectionDescriptor.updatedAt;
             }
 
+            const deletedDocuments = documents.inCollection(collectionId);
+            deletedDocuments.forEach((doc) => {
+              doc.deletedAt = collectionDescriptor.updatedAt;
+              policies.remove(doc.id);
+            });
```

Comment:
This is what `removeCollectionDocuments` should do on L188, no?

Reply:
`removeCollectionDocuments` seems like it removes the documents from the store, rather than marks them as deleted. For example, it would do that when someone loses permissions to a collection rather than this more extreme case of the collection being deleted altogether

Reply:
We're still calling `removeCollectionDocuments` below, will that not just remove the docs modified in this loop?

Reply:
If you're currently looking at one of those documents, removing the collection relationship alone won't quite do the trick. This loop puts those browser windows into the same state as if you had deleted just the single document

Reply:
Understand, works well in testing too